### PR TITLE
[JSX] Resolve all NPM warnings

### PR DESCRIPTION
**Pull request by HenriRabalais**
I'm getting pretty tired of all these NPM warnings for missing jsdocs or lines that exceed 80 characters so I started resolving the warnings. Some of these jsdocs seem superfluous, but hey, that's what it's asking for.

Should I keep going down this path for all jsx files? Y/N?

### CODE REVIEW

**Comment by maltheism** [comment]
@HenriRabalais I agree it would be really nice to witness the day we can compile without those warnings visible.

**Comment by driusan** [comment]
Y.

**Comment by HenriRabalais** [comment]
@johnsaigle In general I agree - but for constructor, lifecycle and render functions there really isn't much to say. I could have maybe a single sentence that is copied throughout the code, but because these are inherited react functions, there's not much to document. I'll see if inheritdoc works for these!

We don't really have any parent classes in our jsx code in the way that php does. Components are sort of used like building blocks that are nested within one another, but we don't really have functions that are redeclared/overwritten by children, they're just passed down by the parent and props and accessed that way.

In terms of the errors, it seems like now that I'm writing the docs I've forgotten about declaring params and returns!

**Comment by maltheism** [comment]
@johnsaigle @HenriRabalais maybe only add a description when there is something complicated going on in the constructor?

**Comment by johnsaigle** [comment]
It might be good to look into solutions for linting that are aware of React structures, if that's the issue we're having. 

Something like this?
https://www.npmjs.com/package/eslint-plugin-react

**Comment by driusan** [comment]
There's nothing React specific about the failures, it's just functions missing the `@param` and `@return` annotations.

**Comment by HenriRabalais** [comment]
@johnsaigle I've updated the comments. 

Let me know if this works for everyone, and I will keep going through the code with the changes!

File: jsx/Breadcrumbs.js [hunk 1]
```diff
@@ -18,32 +6,36 @@ import PropTypes from 'prop-types';
...
 
+  /** {@inheritdoc} */
   componentWillUnmount() {
     if (typeof window !== 'undefined') {
       window.removeEventListener('resize', this.checkScreenSize);
     }
   }
 
+  /**
+   * User to check the current window size and sets the number of breadcrumbs
```
Comment (johnsaigle) [inline jsx/Breadcrumbs.js]:
```suggestion
   * Checks the current window size and sets the number of breadcrumbs
```

File: jsx/DataTable.js [hunk 2]
```diff
@@ -170,6 +195,12 @@ class DataTable extends Component {
     return filteredIndexes;
   }
 
+  /**
+   * Sort of the row indexes.
```
Comment (johnsaigle) [inline jsx/DataTable.js]:
```suggestion
   * Returns the supplied array sorted by indexes.
```

**Pull request closed without merging**